nibabel/fileholders.py

```diff
@@ -87,12 +95,15 @@ def same_file_as(self, other):
         return (self.filename == other.filename) and (self.fileobj == other.fileobj)
 
     @property
-    def file_like(self):
+    def file_like(self) -> str | io.IOBase | None:
         """Return ``self.fileobj`` if not None, otherwise ``self.filename``"""
         return self.fileobj if self.fileobj is not None else self.filename
 
 
-def copy_file_map(file_map):
+FileMap = ty.Mapping[str, FileHolder]
+
+
+def copy_file_map(file_map: FileMap) -> FileMap:
```

Comment:
Definitely nothing critical, but this might be a little more accurate as:

```suggestion
FileMap = ty.Mapping[str, FileHolder]
FileDict = dict[str, FileHolder]


def copy_file_map(file_map: FileMap) -> FileDict:
```

Reply:
I doubt anybody else will bother constructing this type, and if they do it would be wild to use something less convenient than dict.

Reply:
So why not:
```suggestion
FileMap = dict[str, FileHolder]


def copy_file_map(file_map: FileMap) -> FileMap:
```
?

Reply:
Ah right, we can't subscript `dict` until a later version than 3.8 (it's either 3.9 or 3.10) unless we hide it behind a `ty.TYPE_CHECKING` guard.